lib/verify/response.rb

```diff
@@ -42,19 +42,25 @@ def scenario
     private
 
     def address
-      @address ||= parameters.fetch("attributes").fetch("addresses").first.fetch("value")
+      @address ||= most_recent_verified_value(parameters.fetch("attributes").fetch("addresses"))
     end
 
     def address_lines
       @address_lines ||= address.fetch("lines")
     end
 
     def full_name
-      first_name = parameters.fetch("attributes").fetch("firstNames").first.fetch("value")
-      middle_name = parameters.fetch("attributes").fetch("middleNames").first.fetch("value")
-      surname = parameters.fetch("attributes").fetch("surnames").first.fetch("value")
+      first_name = most_recent_verified_value(parameters.fetch("attributes").fetch("firstNames"))
+      middle_name = most_recent_verified_value(parameters.fetch("attributes").fetch("middleNames"))
+      surname = most_recent_verified_value(parameters.fetch("attributes").fetch("surnames"))
 
       [first_name, middle_name, surname].join(" ")
     end
+
+    def most_recent_verified_value(attributes)
+      attributes.sort_by { |attribute| Date.strptime(attribute["to"], "%Y-%m-%d") }
+        .select { |attribute| attribute["verified"] }
```

Comment:
Would it be better to use `find` here, then you can drop the call to `.last` on the next line

Reply:
I've changed this to use `find`. I had to add a `reverse` in there too to make sure we were getting the most recent value.